compiler/qsc_eval/src/backend.rs

```diff
@@ -82,7 +85,11 @@ pub trait Backend {
     fn qubit_allocate(&mut self) -> usize {
         unimplemented!("qubit_allocate operation");
     }
-    fn qubit_release(&mut self, _q: usize) {
+    /// `false` indicates that the qubit was in a non-zero state before the release,
+    /// but should have been in the zero state.
+    /// `true` otherwise. This includes the case when the qubit was in
+    /// a non-zero state during a noisy simulation.
+    fn qubit_release(&mut self, _q: usize) -> bool {
```

Comment:
I am a bit confused by the last sentence of the docstring. If the qubit is in a non-zero state due to simulation noise it returns `true`? Shouldn't it return `false` in that case?

Reply:
Probably you are missing the second line of the first point - 'but should have been in the zero state'. When simulation is noisy, this assumption is no longer true. It could be in the non-zero state before the release and it's not a problem.
That said, I understand the confusion. I'll probably make this simpler.